src/api/v2/quickPickWizard/ContextValueFilter.ts

```diff
@@ -0,0 +1,27 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ResourceModelBase } from '../v2AzureResourcesApi';
+
+export type ContextValueFilter = string | RegExp | (string | RegExp)[];
+
+export function matchesContextValueFilter(resource: ResourceModelBase, filter: ContextValueFilter): boolean {
+    if (!resource?.quickPickOptions) {
+        return false;
+    }
+
+    const filterArray = Array.isArray(filter) ? filter : [filter];
+
+    return filterArray.some(filter => {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        return resource.quickPickOptions!.contexts.some(contextValue => {
+            if (typeof filter === 'string') {
+                return filter === contextValue;

```

Comment:
I thought the idea was that `contexts` was an array of the context values split up. Like `['azResource', 'contextValue1', 'contextValue2']`. So if you wanted to select resources with a context of `azResource;contextValue1` you'd pass in `['azResource', 'contextValue1']`. This works since the ordering doesn't really matter (we just sort it alphabetically).

Reply:
Oh, yeah that's true. I forgot that this is its own model and is not using the `AzExtTreeItems` to get the context values.